File: 206-reverse-linked-list/206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        
        ListNode curr=head;
        ListNode aft=head;
        ListNode pre=null;
        while(curr!=null){
            aft=curr.next;
            curr.next=pre;
            pre=curr;
            curr=aft;
        }
        return pre;
        
    }
}

File: 206-reverse-linked-list/206-reverse-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    let curr=head
    let aft=head
    let pre=null
    while(curr){
        aft=curr.next
        curr.next=pre
        pre=curr
        curr=aft
    }
    return pre;
    
};